src/dispatcher/mod.rs

```diff
@@ -0,0 +1,279 @@
+use crate::clear_console;
+use crate::ItemId;
+use crate::RoomId;
+use std::sync::atomic::{AtomicU8, Ordering};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum ActionId {
+    GoToControlRoom,
+    GoToHall,
+    GoToToneRoom,
+    GoToVocalBooth,
+
+    SpillBeerOnTheConsole,
+
+    PickConsole,
+    PushRandomButton,
+    MakeAllFadersUp,
+    StaringAtTheDeskWithTheDumbFace,
+
+    PickImac,
+    TurnOffImac,
+    BreakScreenOfImac,
+    CloseOpenedProToolsSession,
+    TouchImacDisplayWithFinger,
+
+    PickChair,
+    KickChairFromSoundGuy,
+    PutLegsOnTheChair,
+
+    PickBottle,
+    MakeHoleInBottleAndGetHigh,
+    ThrowBottleToTheAssistant,
+    DrinkFromBottle,
+
+    PickDrums,
+    HitCymbalsRandomly,
+    ScreamAtTheSnare,
+
+    PickGuitarAmp,
+    TurnVolumeToEleven,
+    AllButtonsUp,
+    PressStandBy,
+
+    PickMic,
+    SwearAtTheCleaner,
+    CaughtForThreeMinutesStraight,
+    ThrowMicAtTheWall,
+
+    SingToTheCureAndCry,
+    PressRandomNumbersOnAnIntercom,
+    StaringAtTheDoorForLoongTime,
+
+    DropItem,
+
+    ExitGame,
+}
+
+static MICS_BROKEN: AtomicU8 = AtomicU8::new(0);
```

Comment:
Wondering why you chose to make this static. Probably because you want this to be mutable? Maybe a better option would be `lazy_static`? But now that I see, maybe it's probably better to create a struct to represent the mic, and inside have functions that mutate that struct. Something like:
```rust
struct Mic {
   count: i32,
}

impl Mic {
   fn break_mic() {
   ...
}
}
```